Nicholas fiori dataset viewer to preview first n rows of dataset

Nick-CS-1 opened the pull request:
Features

- Added a dataset viewer.
- Added ability for dataset viewer to show first 20 and last 10 if CSV file contains more than 30 rows.

Testing

- Go to datasets.
- Click on Show This dataset for a previously uploaded dataset.
- Look towards the bottom to see the data preview.
- If the file contains more than 30 rows, it will show a message stating it only shows the first 20 and last 10.

Nick-CS-1 committed:
Created a new page for Inference (WIP)

1. Created new inference page
2. Added link to get to new page
3. Added fields for user input
4. Added link to go back to deployments
5. Added a submit button
6. Added an inference result that will use the fields

Issue #27

Nick-CS-1 committed:
Merge main

1. Merged with main

Issue #27

Nick-CS-1 committed:
Added testing and changed field inputs

1. Changed the age field to only accept integers.
2. Changed the height field to only accept numbers with a decimal point up to 0.01.
3. Changed the car field to be a checkbox.
4. Added Rails testing.

Issue #27

Nick-CS-1 committed:
Added the ability to preview datasets that are uploaded

1. Added ability to preview datasets
2. If there is over 30 rows it will preview the first 20 and last 10

Issue #24

Mohammad4844 commented on mlite_rails_app/app/controllers/deployments_controller.rb:
```diff
@@ -1,37 +1,37 @@
   def edit
     @models = current_user.models
+    @inference_result = nil # Ensure @inference_result is nil on edit page
   end
```
all these `@inference_result = nil` dont do anything here. u can remove them. its nil by default if its never assigned

Mohammad4844 commented on mlite_rails_app/app/controllers/deployments_controller.rb:
```diff
@@ -63,6 +63,31 @@ def destroy
+  # GET /deployments/:id/inference
+  def inference
+    @inference_result = nil # Ensure the form starts fresh without an old inference result
+  end
```
same here

Nick-CS-1 committed:
Reverted deployment files and modified the dataset viewer.

1. Reverted the deployments controller.
2. Reverted the deployments test controller.
3. Reverted the routes.
4. Reverted the deployments index.
5. Added a space in the dataset viewer to separate the first 20 and last 10.
6. Modified the dataset controller to work with the separation.

Issue #24

Nick-CS-1 committed:
Delete inference page

1. Removed the inference page

Issue #24

Nick-CS-1 committed:
Seperated dataset into two tables and added rails testing

1. Added ability for a dataset's first 20 and last 10 to be split up into two tables if the dataset is bigger than 30 rows.
2. Added rails testing for the datasets test controller.

Issue #24

Nick-CS-1 committed:
Seperated dataset into two tables and added rails testing

1. Added ability for a dataset's first 20 and last 10 to be split up into two tables if the dataset is bigger than 30 rows.
2. Added rails testing for the datasets test controller.
3. Removed inference page for deployments.

Issue #24

Nick-CS-1 committed:
Seperated dataset into two tables and added rails testing

1. Added ability for a dataset's first 20 and last 10 to be split up into two tables if the dataset is bigger than 30 rows.
2. Added rails testing for the datasets test controller.
3. Removed inference page for deployments.

Issue #24

Mohammad4844 committed:
Merge main and fix conflicts

Mohammad4844 committed:
Fix an edge case to pass tests

Mohammad4844 reviewed (approved):
Looks good. I merged main and fixed an annoying merge conflict + updated to make tests pass.
